### PR TITLE
SCMOD-13301 Worker-Framework: Move away from Files.exists

### PR DESCRIPTION
**Pull request by simranjeetc**
https://portal.digitalsafe.net/browse/SCMOD-13301

### CODE REVIEW

**Review by dermot-hardy — COMMENTED** [review]
What you've done here is really debugging code for [SCMOD-12970](https://portal.digitalsafe.net/browse/SCMOD-12970) rather than the replacement implementation expected for [SCMOD-13301](https://portal.digitalsafe.net/browse/SCMOD-13301).  We can go ahead and check this in but make sure it is marked up as progressing SCMOD-12970.  For SCMOD-13301 the exception needs to be propagated so that the calling code can decide what to do.

**Comment by simranjeetc** [comment]
> What you've done here is really debugging code for [SCMOD-12970](https://portal.digitalsafe.net/browse/SCMOD-12970) rather than the replacement implementation expected for [SCMOD-13301](https://portal.digitalsafe.net/browse/SCMOD-13301). We can go ahead and check this in but make sure it is marked up as progressing SCMOD-12970. For SCMOD-13301 the exception needs to be propagated so that the calling code can decide what to do.

Correct. This is debugging code and not a replacement. We have not reached there yet. This and the other two PRs
https://github.houston.softwaregrp.net/Verity/deploy/pull/1289
https://github.com/WorkerFramework/worker-framework/pull/128
will NOT be merged. The test can be run by deploying the SCMOD-13301 branch of the deploy project. This was just for review so that we can add any more debugging info if required.

**Review by dermot-hardy — APPROVED** [review]
The level of debugging is fine for that purpose.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java [hunk 1]
```diff
@@ -131,12 +131,7 @@ private void handleSingleBatchId(final String batchId, final BatchWorkerServices
             final BatchId batchIdExtracted = extractBatchIds(batchId).get(0);
             final Path pathOfSubBatches = fileSystemProvider.getPathForBatch(tenantId, batchIdExtracted);
 
-            if (!Files.exists(pathOfSubBatches)) {
-                log.error("Exception while reading the batch: " + batchId + " was not found");
```
Comment (dermot-hardy) [inline worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java:135]: I see you're discarding this log statement.  Double-check that it is not required if you haven't already.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java [hunk 2]
```diff
@@ -192,7 +191,16 @@ private void handleSubbatch(final String subbatch, final BatchWorkerServices bat
             batchWorkerServices.registerItemSubtask(DocumentWorkerConstants.DOCUMENT_TASK_NAME, 1, document);
         }
     }
-
+    
+    private void throwBatchExceptionIfFileDoesNotExist(final String subbatch, final Path subbatchFileName) throws BatchDefinitionException
```
Comment (dermot-hardy) [inline worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java]: - Utility methods should be `static`.
- Consider changing the name to `throwBatchExceptionIfFileIsNotAccessible`.  I think that may be more accurate.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java [hunk 3]
```diff
@@ -153,9 +148,16 @@ private void handleSingleBatchId(final String batchId, final BatchWorkerServices
         } catch (final InvalidBatchIdException | InvalidTenantIdException ex) {
             log.error("Exception while handling single batch id: " + ex.getMessage());
             throw new BatchDefinitionException("Exception while handling a single batch id: " + ex.getMessage());
+        } catch(final IOException ex) {
+            throw new BatchDefinitionException("Exception while reading the batch: " + batchId + " was not found", ex);
         }
     }
-
+    
+    private static void throwIOExceptionIfFileDoesNotExist(final Path path) throws IOException
```
Comment (dermot-hardy) [inline worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java]: Same comment on the name.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java [hunk 4]
```diff
@@ -166,10 +168,7 @@ private void handleSubbatch(final String subbatch, final BatchWorkerServices bat
         log.debug("I am going to read each line of: " + subbatchFileName);
         final List<String> lines = new ArrayList<>();
         try {
-            if (!Files.exists(subbatchFileName)) {
-                log.error("Exception while reading subbatch: " + subbatch + ", it does not exist");
```
Comment (dermot-hardy) [inline worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java:170]: Same comment on this log statement being discarded also.  (I would expect that the framework logs the exceptions so I would expect the removal of these log lines to be fine.)

**Review by dermot-hardy — APPROVED** [review]

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java (hunk 1 above)
Comment (simranjeetc) [reply worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java:135]: Did not see the thrown exceptions being logged in BatchWorker or StreamingWorkerWrapper. So added those log statements back.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java (hunk 4 above)
Comment (simranjeetc) [reply worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java:170]: Did not see the thrown exceptions being logged in BatchWorker or StreamingWorkerWrapper. So added those log statements back.

File: worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java [hunk 5]
```diff
@@ -192,7 +192,17 @@ private void handleSubbatch(final String subbatch, final BatchWorkerServices bat
             batchWorkerServices.registerItemSubtask(DocumentWorkerConstants.DOCUMENT_TASK_NAME, 1, document);
         }
     }
-
+    
+    private static void throwBatchExceptionIfFileIsNotAccessible(final String subbatch, final Path subbatchFileName) throws BatchDefinitionException
```
Comment (dermot-hardy) [inline worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java]: Stay within the 138-character guideline.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**